src/TaskTemplate.php

```diff
@@ -250,6 +287,42 @@ public function displaySpecificTypeField($ID, $field = [], array $options = [])
         }
     }
 
+    public function prepareInputForAdd($input)
+    {
+        $input = parent::prepareInputForAdd($input);
+
+        if ($input === false) { // Vérifier si la validation parente a échoué
+            return false;
+        }
+
+        if (isset($input['users_id_tech']) && (int) $input['users_id_tech'] == -1) {
+            $input['use_current_user'] = 1;
+            $input['users_id_tech'] = 0;
+        } else {
+            $input['use_current_user'] = 0;
+        }
+
+        return $input;
```

Comment:
You can extract the common code from `prepareInputForAdd()` and `prepareInputForUpdate()` in a function so as not to duplicate the code.

AND please put the comments in English.